### ISSUE TITLE
Search returns errors or unexpected results

### ISSUE THREAD

**Issue by advancedthreat** [Graylog2/graylog2-web-interface#1523]
Created extraction for "search_query" (see example syslog message below).  An example value that is extracted properly is:
```
cb.urlver=1&q=netconn_count%3A%5B1%20TO%20*%5D%20digsig_result%3Aunsigned
```
When performed a search, the results are inconsistent and sometimes return errors (even after escaping the special characters).  

This search using the extracted field returns an error:
`CB_search_query:cb.urlver=1\&q=netconn_count%3A%5B1%20TO%20\*%5D%20digsig_result%3Aunsigned`
```
org.apache.lucene.queryparser.classic.ParseException
Cannot parse 'CB_search_query:cb.urlver=1\&q=netconn_count:[1 TO \*] digsig_result:unsigned': Encountered " ":" ": "" at line 1, column 44. Was expecting one of: <EOF> <AND> ... <OR> ... <NOT> ... "+" ... "-" ... <BAREOPER> ... "(" ... "*" ... "^" ... <QUOTED> ... <TERM> ... <FUZZY_SLOP> ... <PREFIXTERM> ... <WILDTERM> ... <REGEXPTERM> ... "[" ... "{" ... <NUMBER> ...
```
This search works as expected, using a wildcard instead of the complete value:  
```
CB_search_query:(cb.urlver=1&q=netconn_count%3*)
```
This search does not work: `CB_search_query:(cb.urlver=1&q=netconn_count%3A*)`

Searching for the full string (escaping special characters) does not work at all.

The full syslog message is shown below.
```
2015-07-13 01:28:10 [29278] <warning>  customer="Critical Start" base_url="xyz.analytics.com" modload_count=104 process_md5=95ebcd689fc41b1b658ac5fb9ce08714 alliance_link_atawhitelist=http://www.ata.com feed_id=2 alliance_updated_bit9endpointvisibility=2014-12-10T20:16:46.000Z crossproc_count=0 ioc_attr={} parent_name=svchost.exe group=Production Critical Start server_name=ip-10-2-1-23 start=2015-07-13T01:12:48.745Z filemod_count=16 parent_id=-4365978491798759270 comms_ip=70.123.127.110 regmod_count=7 interface_ip=192.168.0.13 process_pid=4208 username=Godzilla\Travis process_name=livecomm.exe process_id=0000002a-0000-1070-01d0-bd09086cea74 report_id=f435128d-f5c4-4fa1-9f1f-14d0051ff09c alliance_data_bit9endpointvisibility=f435128d-f5c4-4fa1-9f1f-14d0051ff09c segment_id=1 alliance_score_atawhitelist=100 ioc_value={"index_type": "events", "search_query": "cb.urlver=1&q=netconn_count%3A%5B1%20TO%20*%5D%20digsig_result%3Aunsigned"} cb_version=5.1.0.150625.500 os_type=windows childproc_count=0 sensor_id=42 watchlist_14=2015-07-13T01:20:05.117246Z parent_unique_id=0000002a-0000-02d4-01d0-bc99f8cb3bb5-00000001 cmdline="C:\Program Files\WindowsApps\microsoft.windowscommunicationsapps_17.5.9600.20911_x64__8wekyb3d8bbwe\LiveComm.exe" -ServerName:Microsoft.WindowsLive.Platform.Server ioc_type=query report_title=Unsigned file with network connections hostname=GODZILLA alliance_updated_atawhitelist=2015-07-11T19:38:32.000Z last_update=2015-07-13T01:13:53.357Z feed_name=bit9endpointvisibility netconn_count=1 watchlist_47=2015-07-13T01:20:05.117246Z parent_pid=724 alliance_link_bit9endpointvisibility=https://www.bit9.com/cbfeeds/endpointvisibility_feed.xhtml#9 path=c:\program files\windowsapps\microsoft.windowscommunicationsapps_17.5.9600.20911_x64__8wekyb3d8bbwe\livecomm.exe alliance_data_atawhitelist=['8c9ad98ad4c7350050d3fe249c3cf462'] alliance_score_bit9endpointvisibility=1 host_type=workstation event_timestamp=1436750890.83 unique_id=0000002a-0000-1070-01d0-bd09086cea74-00000001
```

**Comment by edmundoa** [comment]
Hi,

The problem here is that the encoded query you are using to search in Graylog is passed in the URL and decoded in the server. As the search query was not escaped before encoding it, it might be invalid in the Lucene query language, returning the error you see.

We want to start using POST to pass search parameters (see #1463) and that should already fix some problems like this one. In the meantime you could try to use phrases for searching for those complex parameters, i.e. enclosing the query in quotation marks:

```CB_search_query:"cb.urlver=1\&q=netconn_count%3A%5B1%20TO%20\*%5D%20digsig_result%3Aunsigned" ```

**Comment by edmundoa** [comment]
Closing, as this will be fixed once #1463 is fixed.
